django/db/models/deletion.py

```diff
@@ -51,6 +52,23 @@ def DO_NOTHING(collector, field, sub_objs, using):
     pass
 
 
+def get_candidate_relations_to_delete(opts):
+    # Collect models that contain candidate relations to delete. This may include
+    # relations coming from proxy models.
+    candidate_models = {opts}
+    candidate_models = candidate_models.union(opts.concrete_model._meta.proxied_children)
+    # For each model, get all candidate fields.
+    candidate_model_fields = chain.from_iterable(
+        opts.get_fields(include_hidden=True) for opts in candidate_models
+    )
+    # Keep related objects not coming from ManyToManyFields from all candidate
+    # model fields.
+    return (
+        f for f in candidate_model_fields
+        if f.auto_created and not f.concrete and not f.many_to_many
```

Comment:
In this theoretical case, you would want the related objects to recursively delete, right? Shouldn't it be included in this list?

Also, wouldn't you call the "reverse" of a ForeignKey a ManyToOneField? (if a ForeignKey a "one to many")